source/adapters/hip/command_buffer.hpp

```diff
@@ -9,7 +9,222 @@
 //===----------------------------------------------------------------------===//
 
 #include <ur/ur.hpp>
+#include <ur_api.h>
 
-/// Stub implementation of command-buffers for HIP
+#include "context.hpp"
+#include <hip/hip_runtime.h>
+#include <memory>
 
-struct ur_exp_command_buffer_handle_t_ {};
+static inline const char *getUrResultString(ur_result_t Result) {
```

Comment:
Is there no way of sharing this string-ifier, a quick grep through the repo reveals it's been done in 3 other places:
* https://github.com/oneapi-src/unified-runtime/blob/b2c7e83521949cd6039cdd1c3874ed0bbd238939/include/ur_print.hpp#L328
* https://github.com/oneapi-src/unified-runtime/blob/b2c7e83521949cd6039cdd1c3874ed0bbd238939/source/adapters/level_zero/common.hpp#L30
* https://github.com/oneapi-src/unified-runtime/blob/b2c7e83521949cd6039cdd1c3874ed0bbd238939/source/adapters/cuda/command_buffer.hpp#L18

Reply:
This makes sense. Thanks for the suggestions. However, we prefer to keep this PR focusing only on graph support for the HIP backend and open another refactoring PR that will touch all implemented backends.

Reply:
OK, do we need a ticket or will it be handled internally?

Reply:
I'll create an internal ticket to do this

Reply:
Thanks @EwanC 🏅